autoload/go/path.vim

```diff
@@ -134,8 +128,8 @@ function! go#path#BinPath() abort
   elseif $GOBIN != ""
     let bin_path = $GOBIN
   else
-    " GOPATH (user set or default GO)
-    let bin_path = expand(go#path#Default() . "/bin/")
+    let go_paths = split(go#path#Default(), go#util#PathListSep())
```

Comment:
@myitcv this returns the path where the binaries might be installed, not the path of the binary itself. However I assume we can use your approach inside `CheckBinPath()` where it looks for the path of the binary itself